src/main/java/com/atlassian/jira/cloud/jenkins/util/XmlUtils.java

```diff
@@ -0,0 +1,49 @@
+package com.atlassian.jira.cloud.jenkins.util;
+
+import hudson.XmlFile;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.StringReader;
+
+/**
+ * Utility class for handling XML-related operations such as parsing XML files and extracting values
+ * from XML elements.
+ *
+ * <p>This class provides methods to parse XML files, extract specific values based on tag names,
+ * and handle boolean values stored within XML elements.
+ */
+public class XmlUtils {
+
+    public Element parseXmlFile(final XmlFile file)
+            throws ParserConfigurationException, IOException, SAXException {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.parse(new InputSource(new StringReader(file.asString())));
+        return document.getDocumentElement();
+    }
+
+    public String extractXmlValue(final Element parentElement, final String tagName) {
+        NodeList nodeList = parentElement.getElementsByTagName(tagName);
+        if (nodeList.getLength() > 0) {
+            return nodeList.item(0).getTextContent();
+        }
+        return "";
```

Comment:
Instead of returning an empty string here I think we should throw an exception? ` throw new IllegalArgumentException("Tag not found: " + tagName);`

Reply:
"" is the default value we are after, its expected not to find things sometimes, kinda like a 204 httpcode lolol